app/components/documents/document-preview/document-preview-modal.hbs

```diff
@@ -39,6 +39,7 @@
           @unmarkForSignature={{@unmarkForSignature}}
           @didDeletePiece={{@didDeletePiece}}
           @onChangeFile={{perform this.loadFile}}
+          @mayAccessKabinetDocument={{@mayAccessKabinetDocument}}
```

Comment:
this `onChangeFile` was on DEV but htis branch was behind.
We have to recalculate if we can see the file/piece when switching version from the version tab.
I would say the current `loadFile` is not high enough, we have to callback our router/controller and reload piece (file included) to check the access first.